src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -274,36 +274,44 @@ private SNINpHandle CreateNpHandle(string fullServerName, long timerExpire, obje
         {
             if (parallel)
             {
-                SNICommon.ReportSNIError(SNIProviders.INVALID_PROV, 0, SNICommon.MultiSubnetFailoverWithNonTcpProtocol, string.Empty);
+                SNICommon.ReportSNIError(SNIProviders.NP_PROV, 0, SNICommon.MultiSubnetFailoverWithNonTcpProtocol, string.Empty);
                 return null;
             }
 
-            // Named Pipe Format: np:\\<host name>\pipe\<pipe name>
-            const int minPipeLength = 10; // eg: //./pipe/a
-            string pipePath = fullServerName.Trim();
-            if (pipePath.Length < minPipeLength)
+            if(fullServerName.Length == 0 || fullServerName.IndexOf('/') != -1) // Pipe paths only allow back slashes
```

Comment:
I looked it up and Contains is also faster by default, since it uses Ordinal comparison rather than culture-dependent comparisons.